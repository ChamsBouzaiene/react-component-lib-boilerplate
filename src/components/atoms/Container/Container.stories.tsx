import React from 'react';
import { Story, Meta } from '@storybook/react';

import Container, { Props } from './Container';

export default {
    title: 'Atoms/Container',
    component: Container,
} as Meta;

const Template: Story<Props> = (args) => <Container {...args} />;

export const Default = Template.bind({});
Default.args = {
    children: <span>Welcome</span>,
};
