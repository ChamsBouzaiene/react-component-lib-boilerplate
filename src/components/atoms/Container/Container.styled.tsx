import styled from 'styled-components';

export const StyledDiv = styled.div`
    font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 15px 20px;
    display: flex;
    align-items: center;
    justify-content: space-between;
`;
