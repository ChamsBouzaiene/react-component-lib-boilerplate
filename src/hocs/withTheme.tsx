import React from 'react';
import { ThemeProvider } from 'styled-components';
import darkTheme from '../theme';

export function withTheme<P>(WrappedComponent: React.ComponentType<P>): React.FC<P> {
    const ComponentWithTheme = (props: P) => {
        return (
            <ThemeProvider theme={darkTheme}>
                <WrappedComponent {...props} />
            </ThemeProvider>
        );
    };
    return ComponentWithTheme;
}
export default withTheme;
