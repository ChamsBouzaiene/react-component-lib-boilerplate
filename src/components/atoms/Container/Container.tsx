import React from 'react';
import PropTypes from 'prop-types';
import { StyledDiv } from './Container.styled';

export interface Props {
    children: React.ReactNode;
}

const Container: React.FC<Props> = ({ children }) => <StyledDiv>{children}</StyledDiv>;

export default Container;

Container.propTypes = {
    children: PropTypes.node.isRequired,
};
