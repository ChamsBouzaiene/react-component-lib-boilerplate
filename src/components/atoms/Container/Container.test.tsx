import React from 'react';
import { render, screen } from '@testing-library/react';
import Container from './Container';
import { withTheme } from '../../../hocs/withTheme';

const ThemedContainer = withTheme(Container);

test('renders a default Container', () => {
    const { baseElement } = render(<ThemedContainer>TESTING</ThemedContainer>);
    expect(screen.getByText('TESTING')).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
});
