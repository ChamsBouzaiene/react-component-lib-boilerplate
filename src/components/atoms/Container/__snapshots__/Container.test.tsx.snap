// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a default Container 1`] = `
<body>
  <div>
    <div
      class="sc-bdnxRM euCGiU"
    >
      TESTING
    </div>
  </div>
</body>
`;
